src/dreyfus_index_updater.erl

```diff
@@ -31,6 +31,14 @@ update(IndexPid, Index) ->
     erlang:put(io_priority, {view_update, DbName, IndexName}),
     {ok, Db} = couch_db:open_int(DbName, []),
     try
+        {ok, IdxPurgeSeq} = clouseau_rpc:get_purge_seq(IndexPid),
+        FoldFun = fun(PurgeSeq, {Id, _Revs}, Acc) ->
+            clouseau_rpc:delete(IndexPid, Id),
```

Comment:
This `FoldFun` is enum the purged docs, and delete the invalid index. But the index will be rebuilt later.
The purge API is work on id/rev pairs, but one doc has only one index. So if the doc is changed, the index should be delete first and then rebuild it.